std/range/interfaces.d

```diff
@@ -520,6 +538,14 @@ template outputRangeObject(E...) {
      static assert(is(typeof(appWrapped) : OutputRange!(uint)));
 }
 
+/// Thrown when an interface method is not supported by the wrapped range
+class UnsupportedMethod : Exception
```

Comment:
I think that name is too generic to be on `std.range.interfaces`

Reply:
Do you have a suggestion for a better name?

Reply:
I would either:
1. make it range specific, e.g. `UnsupportedRangeMethod` which can be bigger, or `NoRangeMethod`
2. move this exception to `std.exception`

I'm more inclined to the `2.` since:
1. this can be used by other users generically. Not only useful for users unsupported interface methods but sometimes people would throw this if something is not yet implemented on a normal function and want the error to be catchable.
2. more `UnsupportedMethod` may occur in other interface implementation, even inside Phobos and having two `UnsupportedMethod` imports can be annoying to deal by the users, plus duplication will occur without being really necessary.

Reply:
Also, `2.` can allow `1.` in the future by extending it, if really necessary.

Reply:
Phobos generally does not share exceptions between modules, so I think (1) is the better choice here.

Renamed to `UnsupportedRangeMethod`.